README.md
Radiant Bot
===========

Description comming ...


documentation/NEWS.md
# NEWS

All notable changes to this project will be documented in this file.


src/main.py
# ────────────────────imports────────────────────
import discord
from discord.ext import commands
import random
import time
import sys
import os

# ────────────────────token et autres trucs────────────────────
TOKEN = ''
client = commands.Bot(command_prefix='r!')
date = time.strftime("%A %d %B %Y %H:%M:%S")
client.remove_command('help')

random.seed(2)


def generateXP():
    random.randint(2, 100)


# ────────────────────Message de démarrage du bot────────────────────

@client.event
async def on_ready():
    await client.change_presence(activity=discord.Streaming(name="Radiant Bot", url='https://www.twitch.tv/'))
    print('Je suis OK!')
    embed = discord.Embed(title="『:white_check_mark:』Je suis allumé!", color=0xf0f4f6)
    embed.set_footer(text=date)
    channel = client.get_channel(id=695918330338607104)
    await channel.send(embed=embed)


# ────────────────────cogs────────────────────

@client.command()
async def load(ctx, extension):
    client.load_extension(f'cogs.{extension}')


@client.command()
async def unload(ctx, extension):
    client.unload_extension(f'cogs.{extension}')


for filename in os.listdir('cogs'):
    if filename.endswith('.py'):
        client.load_extension(f'cogs.{filename[:-3]}')


# ────────────────────dodo────────────────────


@client.command
@commands.has_permissions(administrator=True)
async def dodo(ctx):
    embed = discord.Embed(title="Bonne nuit!", description='je vé dormir UwU')
    await ctx.send(embed=embed)
    await client.logout


# ────────────────────erreurs────────────────────

@client.event
async def on_command_error(ctx, error):
    if isinstance(error, commands.MissingRequiredArgument):
        date = time.strftime("%A %d %B %Y %H:%M:%S")
        embed = discord.Embed(title="『:warning:』Erreur! Vérifiez les arguments requis!", color=0xfe0b00)
        embed.set_footer(text=date)
        await ctx.send(embed=embed)


@client.event
async def on_MissingPermissions(ctx, error):
    if isinstance(error, commands.MissingRequiredArgument):
        date = time.strftime("%A %d %B %Y %H:%M:%S")
        embed = discord.Embed(title="『:warning:』Erreur! Vous n'avez pas les perms!", color=0xfe0b00)
        embed.set_footer(text=date)
        await ctx.send(embed=embed)


@client.event
async def on_command_error(ctx, error):
    if isinstance(error, commands.CommandNotFound):
        date = time.strftime("%A %d %B %Y %H:%M:%S")
        embed = discord.Embed(title="『:warning:』Erreur! Mauvaise commande!!", color=0xfe0b00)
        embed.set_footer(text=date)
        await ctx.send(embed=embed)


# ────────────────────ping/pong────────────────────

@client.command()
async def ping(ctx):
    embed = discord.Embed(title="『:ping_pong:』 Pong!",
                          description=f"{round(client.latency * 1000)}ms")
    await ctx.send(embed=embed)


src/levels.py
import discord
from discord.ext import commands
import time
import random
import mysql.connector

random.seed(2)

dbusers = mysql.connector.connect(
    host='localhost',
    user='Ade',
    password='soligaleo5',
    database='userslevels',
    auth_plugin='mysql_native_password'
)

date = time.strftime("%A %d %B %Y %H:%M:%S")


class Levels(commands.Cog):

    def __init__(self, client):
        self.client = client

    @staticmethod
    def generateXP():
        return random.randint(1, 10)

    @commands.Cog.listener()
    async def on_ready(self):
        print('Cog Levels ok!')
        print(dbusers)

    @commands.Cog.listener()
    async def on_message(self, message):
        if message.author.bot:
            return

        else:
            xp = Levels.generateXP()
            print(message.author.name + " A reçu " + str(xp) + " xp!")
            cursor = dbusers.cursor()
            cursor.execute("SELECT user_xp FROM users WHERE client_id =" + str(message.author.id))
            result = cursor.fetchall()

            if len(result) == 0:
                print("Utilisateur non présent dans la base de données... Ajout de l'utilisateur...")
                cursor.execute("INSERT INTO users VALUES(" + str(message.author.id) + "," + str(xp) + ")")
                dbusers.commit()
                print("Utilisateur ajouté.")
                print(result)

            else:
                newXP = result[0][0] + xp
                print("Xp total de l'utilisateur après ajout de l'xp: " + str(newXP))
                cursor.execute("UPDATE users SET user_xp = " + str(newXP) + " WHERE client_id = " + str(message.author.id))
                dbusers.commit()
                print("Database mise a jour.")

        if newXP >= 1500:
            cursor.execute("SELECT user_level FROM users WHERE client_id =" + str(message.author.id))
            lvlresult = cursor.fetchall()
            newLevel = lvlresult[0][0] + 1
            print(message.author.name + "Est monté d'un niveau! il est maintenant niveau " + str(newLevel))
            cursor.execute("UPDATE users SET user_level = " + str(newLevel) + " WHERE client_id = " + str(message.author.id))
            cursor.execute("UPDATE users SET user_xp = 0 WHERE client_id = " + str(message.author.id))
            dbusers.commit()
            print("Database mise a jour.")


def setup(client):
    client.add_cog(Levels(client))


RadiantBot/Moderation.py
import discord
from discord.ext import commands
import time

date = time.strftime("%A %d %B %Y %H:%M:%S")


class Moderation(commands.Cog):

    def __init__(self, client):
        self.client = client

    @commands.Cog.listener()
    async def on_ready(self):
        print('Cog Modération ok!')

    # ────────────────────clear────────────────────

    @commands.command()
    @commands.has_permissions(manage_messages=True)
    async def clear(self, ctx, amount=99):
        await ctx.channel.purge(limit=amount + 1)
        embed = discord.Embed(title="Clear", description=f"{amount} messages ont été supprimés!", color=0xf5eded)
        embed.set_footer(text=date)
        await ctx.send(embed=embed, delete_after=5)

    # ────────────────────ban/kick/unban────────────────────

    @commands.command()
    @commands.has_permissions(ban_members=True)
    async def ban(self, ctx, member: discord.Member, *, reason=None):
        await member.ban(reason=reason)
        embed = discord.Embed(title="Ban", description=f"{member} a été banni!", color=0xfe0b00)
        embed.set_footer(text=f"BabelBot {date}")
        await ctx.send(embed=embed)

    @commands.command()
    @commands.has_permissions(kick_members=True)
    async def kick(self, ctx, member: discord.Member, *, reason=None):
        await member.kick(reason=reason)
        embed = discord.Embed(title="Kick", description=f"{member} a été kick !", color=0xfe0b00)
        embed.set_footer(text=f"BabelBot {date}")
        await ctx.send(embed=embed)

    @commands.command()
    @commands.has_permissions(ban_members=True)
    async def unban(self, ctx, *, member):
        banned_users=await ctx.guild.bans()
        member_name, member_discriminator=member.split('#')

        for ban_entry in banned_users:
            user=ban_entry.user

            if (user.name, user.discriminator) == (member_name, member_discriminator):
                await ctx.guild.unban(user)
                embed = discord.Embed(title="Unban", description=f"{user.name}#{user.discriminator} a été unban!", color=0xfe0b00)
                embed.set_footer(text=f"Babel | RP {date}")
                await ctx.send(embed=embed)


def setup(client):
    client.add_cog(Moderation(client))


RadiantBot/Main.py
# ────────────────────imports────────────────────
import discord
from discord.ext import commands
import random
import time
import time
import sys
import os

# ────────────────────token et autres trucs────────────────────
TOKEN = ''
client = commands.Bot(command_prefix='r!')
date = time.strftime("%A %d %B %Y %H:%M:%S")
client.remove_command('help')

random.seed(2)


def generateXP():
    random.randint(2, 100)


# ────────────────────Message de démarrage du bot────────────────────

@client.event
async def on_ready():
    await client.change_presence(activity=discord.Streaming(name="Radiant Bot", url='https://www.twitch.tv/'))
    print('Je suis OK!')
    embed = discord.Embed(title="『:white_check_mark:』Je suis allumé!", color=0xf0f4f6)
    embed.set_footer(text=date)
    channel = client.get_channel(id=695918330338607104)
    await channel.send(embed=embed)


# ────────────────────cogs────────────────────

@client.command()
async def load(ctx, extension):
    client.load_extension(f'cogs.{extension}')


@client.command()
async def unload(ctx, extension):
    client.unload_extension(f'cogs.{extension}')


for filename in os.listdir('cogs'):
    if filename.endswith('.py'):
        client.load_extension(f'cogs.{filename[:-3]}')


# ────────────────────dodo────────────────────


@client.command
@commands.has_permissions(administrator=True)
async def dodo(ctx):
    embed = discord.Embed(title="Bonne nuit!", description='je vé dormir UwU')
    await ctx.send(embed=embed)
    await client.logout


# ────────────────────erreurs────────────────────

@client.event
async def on_command_error(ctx, error):
    if isinstance(error, commands.MissingRequiredArgument):
        date = time.strftime("%A %d %B %Y %H:%M:%S")
        embed = discord.Embed(title="『:warning:』Erreur! Vérifiez les arguments requis!", color=0xfe0b00)
        embed.set_footer(text=date)
        await ctx.send(embed=embed)


@client.event
async def on_MissingPermissions(ctx, error):
    if isinstance(error, commands.MissingRequiredArgument):
        date = time.strftime("%A %d %B %Y %H:%M:%S")
        embed = discord.Embed(title="『:warning:』Erreur! Vous n'avez pas les perms!", color=0xfe0b00)
        embed.set_footer(text=date)
        await ctx.send(embed=embed)


@client.event
async def on_command_error(ctx, error):
    if isinstance(error, commands.CommandNotFound):
        date = time.strftime("%A %d %B %Y %H:%M:%S")
        embed = discord.Embed(title="『:warning:』Erreur! Mauvaise commande!!", color=0xfe0b00)
        embed.set_footer(text=date)
        await ctx.send(embed=embed)


# ────────────────────ping/pong────────────────────

@client.command()
async def ping(ctx):
    embed = discord.Embed(title="『:ping_pong:』 Pong!",
                          description=f"{round(client.latency * 1000)}ms")
    await ctx.send(embed=embed)
